packages/adblocker-webextension-cosmetics/adblocker.ts

```diff
@@ -74,28 +76,47 @@ function handleResponseFromBackground(
     ACTIVE = true;
   }
 
+  // TODO - do we really need setTimeout here? And if so, can we use microtask instead?
+
   // Inject scripts
   if (scripts) {
-    for (const script of scripts) {
-      setTimeout(() => injectScript(script, window.document), 0);
-    }
-  }
-
-  // Normal CSS
-  if (styles && styles.length > 0) {
-    setTimeout(() => injectCSSRule(styles, window.document), 0);
+    setTimeout(() => {
```

Comment:
setTimeout 0 may be deferred by quite a bit. Consider using `postMessage`, `requestAnimationFrame` or `Promise.resolve().then` if you want to get the callback to fire asap.

On the other hand if a tick is needed it's always better to wait for a sideeffect before calling the "callback".